### DISCUSSION TITLE
Does ext-plugin have a downgrade strategy?

### DISCUSSION THREAD

**Discussion by arabot777** [apache/apisix#5833] [Q&A] [answered]
Some functions are realized by using java service through **ext-plugin**. But java service has a **suspended animation status**, in this case the apisix service is unavailable. Does apisix support downgrading ext-plugin plugins?

**Comment by juzhiyuan** [comment]
We could seek help from @tzssangglass when he notices this question 😄

**Reply by tzssangglass** [reply]
> suspended animation status

you mean `Stop The World`?

Auto-downgrade is not yet supported, and this situation should occur with a timeout.

What are your thoughts on this issue.

**Reply by tokers** [reply]
Is it possible to add some `init` phases for ext-plugin so that we can launch APISIX only if the plugin runners are ready.

**Reply by arabot777** [reply]
I think the ext-plugin services provides a health interface when the apisix checks that it is unhealthy,the traffic skips the ext-plugin or does not provide the service. This health interface can also be used for alarm and manual intervention.

**Reply by spacewander** [reply]
What about using passive detection? If the ext-plugin failed to handle the request, we skip it instead of returning 503, when the `allow_degradation` is true.

**Reply by arabot777** [reply]
it's great. ext-plugins supply `allow_degradation` param，this  is a simple and universal way

**Comment by tzssangglass** [comment, marked as answer]
`allow_degradation` is for ext-plugin as a whole. If ext-plugin is used to 1. validate permissions 2 some business implementations, then the request is between can be skipped and cannot be skipped.

**Reply by arabot777** [reply]
Currently, we are preparing to migrate the prod domain name. We are going to adopt this scheme. If pr is too late to optimize, will we provide the development work and merge it into the backbone?

**Reply by tzssangglass** [reply]
Waiting for updates, if you are interested, welcome PR!
